hedera-node/hedera-schedule-service-impl/src/main/java/com/hedera/node/app/service/schedule/impl/handlers/AbstractScheduleHandler.java

```diff
@@ -38,25 +39,25 @@ protected void preHandleScheduledTxn(
             final PreHandleContext context,
             final TransactionBody scheduledTxn,
             final AccountID payerForNested,
-            final PreHandleDispatcher dispatcher) {
-        final var innerContext = context.createNestedContext(scheduledTxn, payerForNested);
-        context.setInnerContext(innerContext);
+            final PreHandleDispatcher dispatcher)
+            throws PreCheckException {
+        final var innerContext =
+                context.createNestedContext(scheduledTxn, payerForNested, UNRESOLVABLE_REQUIRED_SIGNERS);
         final HederaFunctionality scheduledFunction;
         try {
             scheduledFunction = functionOf(scheduledTxn);
         } catch (UnknownHederaFunctionality ex) {
-            innerContext.status(SCHEDULED_TRANSACTION_NOT_IN_WHITELIST);
-            return;
+            throw new PreCheckException(SCHEDULED_TRANSACTION_NOT_IN_WHITELIST);
```

Comment:
This is a good point, the error and response code seem wrong. If we get UnknownHederaFunctionality, something is REALLY, REALLY wrong. But since that is outside the scope of this change (i.e. even thought it doesn't look like it I am trying to avoid making extra changes :)), I encourage you to take appropriate action in your work on scheduled transaction